### PR TITLE
Add support for alternative orthographies to Bliss::Orthography

### PR DESCRIPTION
**Pull request by curufinwe**
Second part of my orthography refactoring series.

This PR adds spans to Bliss::Orthography as a structured representation of a sequence of partial orthographies, supporting both text spans and alternative orthography lists. Updates the corpus parser so main <orth> elements can parse nested `<alternatives><orth>...</orth></alternatives>` structures while preserving plain-text behavior for context orthographies.

Also adds focused Bliss unit tests for span behavior, parser compatibility, empty alternatives, nested alternatives, and context orthography handling.

### CODE REVIEW

File: src/Bliss/Orthography.cc [hunk 1]
```diff
@@ -16,11 +16,47 @@
...
     std::string normalized(text);
     Core::normalizeWhitespace(normalized);
     Core::enforceTrailingBlank(normalized);
     return Orthography(normalized);
 }
 
+Orthography::Orthography(std::string const& text) {
+    setNormalized(text);
+}
+
```
Comment (larissakl) [inline src/Bliss/Orthography.cc:43]: I would prefer to have the constructor before the factory functions

**Review by larissakl — APPROVED** [review]

File: src/Bliss/CorpusParser.cc [hunk 2]
```diff
@@ -136,6 +137,153 @@ void OrthographyElement::characters(const char* ch, int len) {
     verify(isWhitespaceNormalized(product_, tolerateTrailingBlank));
 }
 
+class Bliss::OrthographyElement : public Core::XmlParentElement {
+public:
+    typedef Orthography const& HandlerArgument;
+    typedef void (XmlContext::*Handler)(HandlerArgument);
```
Comment (SimBe195) [inline src/Bliss/CorpusParser.cc]:
```suggestion
    using HandlerArgument = Orthography const&;
    using Handler = void (XmlContext::*)(HandlerArgument);
```

File: src/Bliss/CorpusParser.cc [hunk 3]
```diff
@@ -136,6 +137,153 @@ void OrthographyElement::characters(const char* ch, int len) {
...
+    std::string pendingText_;
+
+    void flushText(bool final);
+    void appendAlternatives(std::vector<Orthography> const& alternatives);
+};
+
+class Bliss::OrthographyElement::AlternativesElement : public Core::XmlParentElement {
+public:
+    typedef std::vector<Orthography> const& HandlerArgument;
+    typedef void (XmlContext::*Handler)(HandlerArgument);
```
Comment (SimBe195) [inline src/Bliss/CorpusParser.cc]:
```suggestion
    using HandlerArgument = std::vector<Orthography> const&;
    using Handler = void (XmlContext::*)(HandlerArgument);
```

File: src/Bliss/Orthography.hh [hunk 4]
```diff
@@ -24,33 +25,88 @@ namespace Bliss {
...
+        }
 
+    private:
+        Type type_;
+
+        std::string              text_;
+        std::vector<Orthography> alternatives_;
+    };
+
+    typedef std::vector<Span> SpanList;
```
Comment (SimBe195) [inline src/Bliss/Orthography.hh]:
```suggestion
    using SpanList = std::vector<Span>;
```

File: src/Bliss/Orthography.hh [hunk 5]
```diff
@@ -24,33 +25,88 @@ namespace Bliss {
...
+        std::vector<Orthography> const& alternatives() const {
+            require(type_ == Type::alternatives);
+            return alternatives_;
+        }
 
+    private:
+        Type type_;
+
+        std::string              text_;
+        std::vector<Orthography> alternatives_;
```
Comment (SimBe195) [inline src/Bliss/Orthography.hh]: Since the class always holds either `text_` or `alternatives_` but never both, this could be accomplished via `std::variant` instead.

File: src/Test/Bliss_Orthography.cc [hunk 6]
```diff
@@ -0,0 +1,219 @@
+/** Copyright 2020 RWTH Aachen University. All rights reserved.
```
Comment (SimBe195) [inline src/Test/Bliss_Orthography.cc]:
```suggestion
/** Copyright 2026 RWTH Aachen University. All rights reserved.
```

**Review by SimBe195 — APPROVED** [review]
Again just minor nitpicks.

**Review by hannah220 — APPROVED** [review]
